apiserver/pkg/model/converter.go

```diff
@@ -162,6 +167,10 @@ func PopulateHeadNodeSpec(spec v1alpha1.HeadGroupSpec) *api.HeadGroupSpec {
 		headNodeSpec.Labels = spec.Template.Labels
 	}
 
+	if *spec.EnableIngress {
```

Comment:
If enableIngress is missing, server crashes with `panic: runtime error: invalid memory address or nil pointer dereference` error.
```suggestion
	if spec.EnableIngress != nil && *spec.EnableIngress {
```

Reply:
Also I think it's better to implement 1 feature per PR. It's easier to review and merge.

Reply:
I know, sorry, was in a rush